### PR TITLE
TIG-2836 hide genny internal metrics by default

### PR DESCRIPTION
**Pull request by vrachev**
[Patch showing (for example) ](https://evergreen.mongodb.com/task/sys_perf_linux_3_node_replSet_insert_remove_patch_f4054ff40c1c8309eefdbc87e84e8e8403e51281_603c10a67742ae2eaf0c43fb_21_02_28_21_56_21##hidden=Genny-Setup%252CHelloWorld-DefaultMetricsName.0%252CHelloWorld-DefaultMetricsName.1%252CHelloWorld-DefaultMetricsName.2%252CInsertRemoveTest.Insert%252CInsertRemoveTest.Remove%252CInsertRemove.Setup%252CInsertRemove.ActorStarted%252CInsertRemove.ActorFinished%252Cfio%252Ccpu_noise%252Ciperf%252Ccanary_InsertRemove.ActorFinished%252Ccanary_InsertRemove.Setup%252CNetworkBandwidth%252Cfio_latency_test_write_clat_mean%252Cfio_latency_test_read_clat_mean%252Cfio_iops_test_write_iops%252Cfio_iops_test_read_iops%252Cfio_streaming_bandwidth_test_write_iops%252Cfio_streaming_bandwidth_test_read_iops%252Ccanary_server-cpuloop-10x%252Ccanary_client-cpuloop-1x%252Ccanary_client-cpuloop-10x%252Ccanary_server-sleep-10ms%252Ccanary_ping%252Cinsert_remove%252Ccanary_InsertRemove.ActorStarted&comparehashes=f4054ff40c1c8309eefdbc87e84e8e8403e51281&threads=maxonly&selected.NetworkBandwidth=MAX&selected.fio_latency_test_write_clat_mean=MAX&selected.fio_latency_test_read_clat_mean=MAX&selected.fio_iops_test_write_iops=MAX&selected.fio_iops_test_read_iops=MAX&selected.fio_streaming_bandwidth_test_write_iops=MAX&selected.fio_streaming_bandwidth_test_read_iops=MAX&selected.canary_server-cpuloop-10x=MAX&selected.canary_client-cpuloop-1x=MAX&selected.canary_client-cpuloop-10x=MAX&selected.canary_server-sleep-10ms=MAX&selected.canary_ping=MAX&selected.iperf=MAX&selected.fio=MAX&selected.cpu_noise=MAX&selected.InsertRemoveTest.Insert=MAX&selected.InsertRemoveTest.Remove=MAX&selected.canary_InsertRemove.ActorFinished=MAX&selected.canary_InsertRemove.ActorStarted=MAX&selected.canary_InsertRemove.Setup=MAX&selected.insert_remove=MAX)`InsertRemove.ActorFinished` -> `canary_InsertRemove.ActorFinished`

### CODE REVIEW

File: src/driver/src/DefaultDriver.cpp [hunk 1]
```diff
@@ -70,7 +70,7 @@ void reportMetrics(genny::metrics::Registry& metrics,
                    bool success,
                    metrics::clock::time_point startTime) {
     auto finishTime = metrics::clock::now();
-    auto actorSetup = metrics.operation(workloadName, "Setup", 0u);
+    auto actorSetup = metrics.operation("canary_" + workloadName, "Setup", 0u);
```
Comment (guoyr) [inline src/driver/src/DefaultDriver.cpp:73]: Could you add a detailed comment here explaining why the `canary` prefix is necessary? Also link the perf trend chart regex for the `canary_` prefix.

**Review by guoyr — APPROVED** [review]
LGTM % suggestion

File: src/driver/src/DefaultDriver.cpp (hunk 1 above)
Comment (vrachev) [reply src/driver/src/DefaultDriver.cpp:73]: Sure

**Comment by vrachev** [comment]
evergreen merge

**Comment by vrachev** [comment]
evergreen merge

**Comment by vrachev** [comment]
@rtimmons @aggrand Any thoughts on why `t_integration_test_single_node_replset` is failing when running the commit queue but succeeding in patch builds? I've now had 3 patch builds succeed and 3 commit queue tasks fail with `distutils.errors.CompileError: command 'gcc' failed with exit status 1`

**Comment by vrachev** [comment]
evergreen merge

**Comment by vrachev** [comment]
evergreen merge

**Pull request merged**